tc/sgx/trusted_worker_manager/enclave_wrapper/file_io_processor.cpp

```diff
@@ -0,0 +1,298 @@
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            return FAILED;
+    }
+
+    resultStr = "FILE CLOSE SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+    return SUCCESS;
+}
+
+// Reads file, store content of file in buffer and updates status
+// SUCCESS or FAILED in result parameter
+uint32_t FileRead(std::string fileName, uint8_t *result, size_t resultSize,
+    uint8_t *outBuf, size_t outBufSize) {
+    string resultStr;
+    if (fileName.empty()) {
+        tcf::Log(TCF_LOG_ERROR, "Filename is empty\n");
+        resultStr = "FAILED TO READ: Filename is empty";
+        resultSize = resultStr.length();
+        result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+        return FAILED;
+    }
+
+    try {
+        ifstream fileStream (fileName);
+        if (!fileStream.is_open())
+        {
+            tcf::Log(TCF_LOG_ERROR, "Unable to open file %s\n", fileName.c_str());
+            resultStr = "FAILED TO READ: Unable to open file";
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            fileStream.close();
+            return FAILED;
+        }
+
+        fileStream.seekg (0, fileStream.end);
+        int fileLen = fileStream.tellg();
+        fileStream.seekg (0, fileStream.beg);
+
+        char *buff = new char[fileLen];
+        fileStream.read(buff, fileLen);
+
+        if ( !fileStream) {
+            tcf::Log(TCF_LOG_ERROR, "Failed to read complete file %s\n", fileName.c_str());
+            resultStr = "FAILED TO READ: Couldn't read complete file";
+            resultSize = resultStr.length();
+            result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+            fileStream.close();
+            return FAILED;
+        }
+
+       tcf::Log(TCF_LOG_ERROR, "File %s read has been completed\n", fileName.c_str());
+
+       // Copy the read file content to output buffer
+       outBuf = (uint8_t *)strncpy((char *)outBuf, buff, fileLen);
+       outBufSize = fileLen;
+       delete[] buff;
+       fileStream.close();
+
+    } catch (std::ifstream::failure e) {
+         tcf::Log(TCF_LOG_ERROR, "Caught exception while Reading file: %s\n", e.what());
+         resultStr = "FAILED TO READ: Caught Exception while reading file";
+         resultSize = resultStr.length();
+         result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+         return FAILED;
+    }
+
+    resultStr = "FILE READ SUCCESS";
+    resultSize = resultStr.length();
+    result = (uint8_t *)strncpy((char *)result, resultStr.c_str(), resultSize);
+
+    return SUCCESS;
+}
+
+// Writes file with the content of the buffer and updates status
+// SUCCESS or FAILED in result parameter
+uint32_t FileWrite(std::string fileName, uint8_t *result, size_t resultSize,
```

Comment:
What is the purpose of the `resultSize` parameter? It is unused. The incoming parameter value is always overwritten in the function and the incoming value is never used.
You should test that the string copied to `result` never exceeds the input value of `resultSize` (including the `NULL` character).
Also document in in the function comments that `resultSize` is the maximum size of the `result` buffer.

Reply:
I agree, we need to use resultSize as max threshold value of the string value that can be copied to result parameter. 
I will add this change along with documentation.

Reply:
Remove `resultSize` for `FileWrite() ` (above) as the buffer is not modified. It is needed only for `FileRead()`, not `FileWrite()`

Reply:
result buffer is modified in all file operations since its a placeholder for all file i/o operations. Hence we do need resultSize.